src/components/SyncView.js

```diff
@@ -26,50 +27,69 @@ var messages = {
 export default class SyncView extends React.Component {
   constructor (props) {
     super(props)
-    var self = this
     var ready = 'replication-ready'
     this.state = {
       message: messages[ready],
-      status: ready
+      status: ready,
+      targets: []
     }
-
-    ipcRenderer.on('select-file', function (event, file) {
-      if (!file) return
-      self.selectFile(file)
-    })
   }
 
   onClose () {
     this.props.changeView(MapEditor)
   }
 
-  selectFile (filename) {
+  onError (err) {
+    this.setState({
+      message: 'Error: ' + err.message,
+      status: 'replication-error'
+    })
+  }
+
+  replicate (target) {
     var self = this
-    replicate.start(filename, function (err, res, body) {
+    if (!target) return
+    this.destroyed = false
+    this.stream = replicate.start(target)
+    this.stream.on('data', function (data) {
       if (self.destroyed) return
-      if (err) {
-        self.setState({
-          message: 'Error: ' + err.message,
-          status: 'replication-error'
-        })
-      } else self.setState({status: 'replication-progress'})
+      var row = JSON.parse(data)
+      var status = row.topic
+      if (status === 'replication-error') return this.onError(new Error(row.message))
+      var message = messages[status] || row.message
+      self.setState({message, status})
+    })
+    this.stream.on('done', function () {
+      console.log('stream over')
+    })
+    this.stream.on('error', function (err) {
+      console.error(err)
     })
   }
 
   componentWillUnmount () {
     this.destroyed = true
     this.stream.destroy()
+    clearInterval(this.interval)
+    ipcRenderer.removeListener('select-file', this.selectFile.bind(this))
+  }
+
+  selectFile (event, filename) {
+    if (!filename) return
+    this.replicate({filename})
```

Comment:
double parens?

Reply:
sorry I'm not sure I understand

Reply:
Oh whoops I couldn't see that you were doing {}s and not ()s. Fancy new syntax!